### DISCUSSION TITLE
How Vanna determines which fields to use in a query?

### DISCUSSION THREAD

**Discussion by ifuchs** [vanna-ai/vanna#227] [Q&A]
I want to try to use Vanna to query an SQL Server db. I know I can feed it the DDL but if the field names are less than obvious, how can Vanna produce a query from natural language? Is this something it can learn from queries/responses used for training it? If the "meaning" of a status code in a record is a number which refers into the table of status codes, will Vanna 'know"to use that code table to write sql to query another table?

**Comment by zainhoda** [comment]
Usually it learns pretty quickly from the example queries but if the column names have no semantic meaning, then you're likely going to have to give some descriptions either in comments in the DDL, or as independent pieces of documentation (i.e. column XYZ is defined as ABC).

For your use case, I think GPT-4 will be absolutely necessary because you're going to need more of its "reasoning" capabilities. The Vanna free tier won't work well since it uses `gpt-3.5-turbo`
